### DISCUSSION TITLE
粒子模擬

### DISCUSSION THREAD

**Discussion by gigitilto** [1101-GS4719-C/Python-Community-GS4719-C#2162] [General]
粒子模擬（Particle Simulation）是一種數值計算方法，用於模擬和分析由大量粒子組成的系統的行為與互動。這些粒子可以是物理系統中的原子、分子、顆粒、天體等。

粒子模擬的基本概念
1. 粒子：
- 在模擬中，每個粒子通常具有質量、位置、速度等性質。這些屬性可以隨著時間進行更新，以模擬粒子在力場中的運動。
2. 相互作用：
- 粒子之間的相互作用是粒子模擬的核心。例如，在氣體模擬中，分子之間的碰撞和引力可以影響其運動。
3. 時間演化：
- 模擬通常會根據某個時間步長進行離散化，通過數值方法（如歐拉法或更高階的數值積分方法）來更新粒子的運動狀態。

在我之前學習的過程中，一開始是將粒子與粒子間的相互作用在兩個粒子距離足夠近時進行判斷，但這麼做必須要實時判定所有粒子和所有粒子之間的距離，計算量高達n^2，若要讓結果足夠準確，設定的時間布長必須足夠小，在這個情況下，電腦能夠負荷的粒子僅僅只能1000個，但實際上並不需要計算和自己距離太遠的粒子是否會有相互作用，所以可以藉由判定粒子周圍的其他粒子判斷距離即可，我使用的方法是對空間進行網格化，將粒子定位，便可以找到每個粒子鄰近的粒子，大大減少計算量，讓電腦可以運算超過10000顆粒子。

**Comment by Jinyuu17** [comment]
大家對粒子模擬有興趣的可以聯絡物理系做電漿研究或模擬的老師，可以學到很多，而且還可以用到國家高速網路計算中心的超級電腦幫你運算，非常的有趣。好幾億的東西交給你使用，說出來超有牌面的啦XD
